makefile_components/base_build_go.mak

```diff
@@ -164,6 +164,7 @@ version:
 	@echo VERSION:$(VERSION)
 
 clean: container-clean bin-clean
+	go clean -cache || echo "You're not running latest golang locally" # Make sure the local go cache is clean for testing
```

Comment:
Yes, I'd say that's true. Not sure how it can fail otherwise. I just didn't want it to break people who hadn't updated go yet locally. This is what currently happens if you have golang 1.9 and do a `make clean`:

```
$ make clean
go clean -cache || echo "You're not running latest golang locally" # Make sure the local go cache is clean for testing
flag provided but not defined: -cache
usage: clean [-i] [-r] [-n] [-x] [build flags] [packages]

Clean removes object files from package source directories.
The go command builds most objects in a temporary directory,
so go clean is mainly concerned with object files left by other
tools or by manual invocations of go build.

Specifically, clean removes the following files from each of the
source directories corresponding to the import paths:

	_obj/            old object directory, left from Makefiles
	_test/           old test directory, left from Makefiles
	_testmain.go     old gotest file, left from Makefiles
	test.out         old test log, left from Makefiles
	build.out        old test log, left from Makefiles
	*.[568ao]        object files, left from Makefiles

	DIR(.exe)        from go build
	DIR.test(.exe)   from go test -c
	MAINFILE(.exe)   from go build MAINFILE.go
	*.so             from SWIG

In the list, DIR represents the final path element of the
directory, and MAINFILE is the base name of any Go source
file in the directory that is not included when building
the package.

The -i flag causes clean to remove the corresponding installed
archive or binary (what 'go install' would create).

The -n flag causes clean to print the remove commands it would execute,
but not run them.

The -r flag causes clean to be applied recursively to all the
dependencies of the packages named by the import paths.

The -x flag causes clean to print remove commands as it executes them.

For more about build flags, see 'go help build'.

For more about specifying packages, see 'go help packages'.
You're not running latest golang locally
```